### DISCUSSION TITLE
Mysterious Segmentation Fault

### DISCUSSION THREAD

**Discussion by mclkov** [ggml-org/llama.cpp#3251] [General]
Hello, everyone, I was wondering to ask how was fixed (if it's resolved) "Segmentation Fault" error ? It appeared after a couple of messages in the chat-mode (regardless of -n, repeat n running parameters, running llama 1 models).

I haven't followed this repo for months, I just wonder "what was wrong" in the early version (back when "ggml" model format was a thing, in March ?).

I checked commits, but couldn't find any mentioning of "segmentations":
git log -g --grep=Segmentation
git log -g --grep=segmentation

Big fan of what you guys accomplished, thank you !

**Comment by KerfuffleV2** [comment]
This is a bit of a hard question since various types of problems can lead to a segfault. You'd probably be better off searching issues or discussions with more specific keywords relating to the problem. A lot of the time, issues usually only affect a small number of people because of their hardware, what they're doing, etc.

In the case where an issue is affecting most users, you can expect it to get fixed very quickly. In other words, the commit that caused the problem will be followed by one that fixes it in a very short time frame. Most of the time, testing before accepting pulls avoids that situation though.

**Comment by mclkov** [comment]
Hello, Kerfuffle, I appreciate your help.

I looked around commits made back in those days, and indeed, there's one :)

commit 4870e455b3653f7d7769fa5772b2c90ffad088df (tag: master-4870e45)
Author: Georgi Gerganov <ggerganov@gmail.com>
Date:   Fri Mar 24 00:11:53 2023 +0200

    Fix memory allocation issues and seg faults

There's a twist though:
commit 3cd8dde0d1357b7f11bdd25c45d5bf5e97e284a0 (tag: master-3cd8dde)
Author: Georgi Gerganov <ggerganov@gmail.com>
Date:   Fri Mar 24 06:22:28 2023 +0200

    Revert "Fix memory allocation issues and seg faults"
    
    This reverts commit 4870e455b3653f7d7769fa5772b2c90ffad088df.
    
    Will provide the correct fix later

**Reply by KerfuffleV2** [reply]
If you move forward a bit through the commits you'll probably find the "correct fix" I'd guess. If you can look up whether there's a pull request associated with the request or find issues that mentioned those commits or the pull then you can probably find out what commit actually closed them.
